pkg/storkctl/schedulepolicy.go

```diff
@@ -60,6 +67,218 @@ func newGetSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Interval = &intervalPolicy
+			case "Daily":
+				var dailyPolicy storkv1.DailyPolicy
+				dailyPolicy.Time = time
+				dailyPolicy.ForceFullSnapshotDay = dailyForceFullSnapshotDay
+				if retain == 0 {
+					dailyPolicy.Retain = storkv1.DefaultDailyPolicyRetain
+				} else {
+					dailyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := dailyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Daily = &dailyPolicy
+			case "Weekly":
+				var weeklyPolicy storkv1.WeeklyPolicy
+				weeklyPolicy.Time = time
+				weeklyPolicy.Day = weeklyDay
+				if retain == 0 {
+					weeklyPolicy.Retain = storkv1.DefaultWeeklyPolicyRetain
+				} else {
+					weeklyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := weeklyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Weekly = &weeklyPolicy
+			case "Monthly":
+				var monthlyPolicy storkv1.MonthlyPolicy
+				monthlyPolicy.Time = time
+				monthlyPolicy.Date = monthlyDate
+				if retain == 0 {
+					monthlyPolicy.Retain = storkv1.DefaultMonthlyPolicyRetain
+				} else {
+					monthlyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := monthlyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Monthly = &monthlyPolicy
+			}
+			var schedulePolicy storkv1.SchedulePolicy
+			schedulePolicy.Name = schedulePolicyName
+			schedulePolicy.Policy = policyItem
+			_, err := storkops.Instance().CreateSchedulePolicy(&schedulePolicy)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			msg := fmt.Sprintf("SchedulePolicy %v created successfully", schedulePolicy.Name)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+	//Picking up user inputs and setting the defaults for flags
+	createSchedulePolicyCommand.Flags().StringVarP(&schedulePolicyType, "policy-type", "t", "Interval", "Select Type of Schedule Policy to apply. Interval / Daily / Weekly / Monthly. ")
+	createSchedulePolicyCommand.Flags().IntVarP(&intervalMinutes, "interval-minutes", "", 30, "Specify the interval, in minutes, after which Portworx should trigger the operation. ")
+	createSchedulePolicyCommand.Flags().IntVarP(&retain, "retain", "", 0, "For backup operations, specify how many backups Portworx should retain. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&time, "time", "", "12:00AM", "Specify the time of the day in the 12 hour AM/PM format, when Portworx should trigger the operation. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&dailyForceFullSnapshotDay, "force-full-snapshot-day", "", "MONDAY", "For daily scheduled backup operations, specify on which day to trigger a full backup. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&weeklyDay, "weekly-day", "", "Sunday", "Specify the day of the week when Portworx should trigger the operation. You can use both the abbreviated or the full name of the day of the week")
```

Comment:
Named the flags as weekly-day , monthly-date, interval-minutes because these fields are applicable only for weekly, monthly and interval policies respectively.
I feel it makes the flags more explanatory for the customers

Reply:
agreed.

Reply:
will address in the revision